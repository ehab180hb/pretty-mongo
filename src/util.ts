/**
 * Will check whether or not the given object is empty
 */
export const isEmptyObject = (obj: object) => {
  return obj && obj.constructor === Object && Object.keys(obj).length === 0
}

/**
 * Will remove the fields that do not meet the given condition
 */
export const removeValues = (o: object, condition: (val: any) => boolean) => {
  const newObj = {}
  for (const k in o) if (!condition(o[k])) newObj[k] = o[k]
  return newObj
}

/**
 * Returns an array with ranges from arg 1 up to (but not including) arg 2
 * @example
 * getRange(3, 6) // [ 3, 4, 5 ]
 */
export const getRange = (from: number, to: number): number[] => {
  if (from > to) throw new Error(`Value of 'from' should be lower than or equal to 'to'`)
  return Array.from({ length: to - from }, (_v, k) => k + from)
}
