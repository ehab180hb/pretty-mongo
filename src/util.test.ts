import { removeValues, isEmptyObject, getRange } from './util'

describe('isEmptyObject()', () => {
  test('When the object is empty', async () => {
    expect(isEmptyObject({})).toBe(true)
  })
  test('When the object is not empty', async () => {
    expect(isEmptyObject({ k: 'v' })).toBe(false)
  })
})

describe('removeValues()', () => {
  const testObj = {
    x: {},
    y: { a: 'a' },
  }
  test('Remove values that meet the given condition', async () => {
    expect(removeValues(testObj, isEmptyObject)).toMatchSnapshot()
  })
})

test('Generates proper ranges', () => {
  expect(getRange(1, 6)).toMatchSnapshot()
  expect(getRange(0, 1)).toMatchSnapshot()
  expect(() => getRange(100, 5)).toThrowErrorMatchingSnapshot()
  expect(() => getRange(7, 5)).toThrowErrorMatchingSnapshot()
})
