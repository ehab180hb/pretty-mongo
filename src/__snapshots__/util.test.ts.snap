// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generates proper ranges 1`] = `
Array [
  1,
  2,
  3,
  4,
  5,
]
`;

exports[`Generates proper ranges 2`] = `
Array [
  0,
]
`;

exports[`Generates proper ranges 3`] = `"Value of 'from' should be lower than or equal to 'to'"`;

exports[`Generates proper ranges 4`] = `"Value of 'from' should be lower than or equal to 'to'"`;

exports[`removeValues() Remove values that meet the given condition 1`] = `
Object {
  "y": Object {
    "a": "a",
  },
}
`;
